lib/active_record/connection_adapters/cockroachdb/referential_integrity.rb

```diff
@@ -39,16 +39,12 @@ def disable_referential_integrity
 
           begin
             foreign_keys.each do |foreign_key|
-              begin
-                add_foreign_key(foreign_key.from_table, foreign_key.to_table, **foreign_key.options)
-              rescue ActiveRecord::StatementInvalid => error
-                if error.cause.class == PG::DuplicateObject
-                  # This error is safe to ignore because the yielded caller
-                  # already re-added the foreign key constraint.
-                else
-                  raise error
-                end
-              end
+              # Avoid having PG:DuplicateObject error if a test is ran in transaction.
+              # TODO: verify that there is no cache issue related to running this (e.g: fk
+              #   still in cache but not in db)
+              next if foreign_key_exists?(foreign_key.from_table, name: foreign_key.options[:name])
+
+              add_foreign_key(foreign_key.from_table, foreign_key.to_table, **foreign_key.options)
```

Comment:
@rafiss any idea on the state of https://github.com/cockroachdb/cockroach/issues/19444 ? This would be quite helpful. Otherwise, I think there is some overall work on this because some fk are not correctly deleted...

Reply:
unfortunately we aren't able to prioritize work on https://github.com/cockroachdb/cockroach/issues/19444 right now.

Reply:
https://github.com/cockroachdb/activerecord-cockroachdb-adapter/issues/309